src/sctools/metrics/classes/gene_metrics.py

```diff
@@ -0,0 +1,102 @@
+from .base_metric import BaseMetric
+from collections import Counter, defaultdict, namedtuple
+
+# fragment could also just be a numpy structured array of length 5 with
+# dtype (int32, int8, bool, bool, bool)
+Fragment = namedtuple('Fragment', ['observations', 'min_mismatches', 'is_unique', 'is_spliced'])
```

Comment:
Sorry my comment was out of date. I was thinking it would be possible to store the fragment as a structured array _instead_ of a namedtuple. You are correct about namedtuple. 

Incidentally, namedtuple is designed to be a lightweight named container class, as it's recognized that python classes have unwanted overhead. 

For the fields listed above, the structured datatype could be (uint32, uint8, bool, bool)